### PR TITLE
feat: Introduce `kraft cloud deploy`

### PR DESCRIPTION
**Pull request by nderjung**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

This PR introduces a new subcommand, `kraft cloud deploy` which is aimed at simplifying the deployment of an application by accumulating the packaging process and creation process of an instance on [kraft.cloud](https://kraft.cloud).

- gomod(cloud): Bump sdk.kraft.cloud to v0.3.0
- feat(pkg): Wrap construction of rootfs with processtree
- feat(initrd): Do not use "fancy" buildkit logging
- feat(deploy): Introduce `kraft cloud deploy`

### CODE REVIEW

File: internal/cli/kraft/cloud/instance/create/create.go [hunk 1]
```diff
@@ -25,110 +27,119 @@ import (
...
-				return fmt.Errorf("invalid external port: %w", err)
+				port = split[0]
 			}
-		} else {
-			port, err := strconv.Atoi(opts.Port)
-			if err != nil {
-				return fmt.Errorf("could not parse port number: %w", err)
+
+			if strings.ContainsRune(port, ':') {
+				ports := strings.Split(port, ":")
```
Comment (craciunoiuc) [inline internal/cli/kraft/cloud/instance/create/create.go:121]: `SplitN(port, ":", 2)`

File: internal/cli/kraft/cloud/deploy/deploy.go [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+		}
+
+		break
+	}
+
+	dns := instance.DNS
+	if len(dns) > 0 {
+		for _, port := range opts.Ports {
+			if strings.HasPrefix(port, "443") {
+				dns = "https://" + dns
```
Comment (craciunoiuc) [inline internal/cli/kraft/cloud/deploy/deploy.go:267]: You should check if it already has a prefix

File: internal/cli/kraft/cloud/deploy/deploy.go [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+			}
+		}
+	}
+
+	var color func(...string) string
+	if instance.Status == "running" {
+		color = textGreen
+	} else {
+		color = textRed
+	}
```
Comment (craciunoiuc) [inline internal/cli/kraft/cloud/deploy/deploy.go]: what other states are there? We could also use yellow

File: internal/cli/kraft/cloud/deploy/deploy.go [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+		fmt.Fprintf(iostreams.G(ctx).Out, "       %s: %s\n", textGray("dns"), dns)
+	}
+	fmt.Fprintf(iostreams.G(ctx).Out, " %s: %.2f ms\n", textGray("boot time"), float64(instance.BootTimeUS)/1000)
+	fmt.Fprintf(iostreams.G(ctx).Out, "    %s: %d MiB\n", textGray("memory"), instance.MemoryMB)
+	fmt.Fprintf(iostreams.G(ctx).Out, "      %s: %s\n\n", textGray("args"), strings.Join(instance.Args, " "))
+
+	if instance.Status != "running" && !opts.NoStart {
+		log.G(ctx).Info("it looks like the instance did not come online, to view logs run:")
+		fmt.Fprintf(iostreams.G(ctx).Out, "\n\tkraft cloud inst logs %s", instance.UUID)
+	}
```
Comment (craciunoiuc) [inline internal/cli/kraft/cloud/deploy/deploy.go]: `out := iostreams.G(ctx).Out`

and then use that for shorter lines

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
Some comments from my side

File: internal/cli/kraft/cloud/deploy/deploy.go (hunk 2 above)
Comment (nderjung) [reply internal/cli/kraft/cloud/deploy/deploy.go:267]: The API does not return a URI scheme

File: internal/cli/kraft/cloud/instance/create/create.go (hunk 1 above)
Comment (nderjung) [reply internal/cli/kraft/cloud/instance/create/create.go:121]: In fact, `SplitN` will guarantee 2 always if there is at least 1 `:`. The existing `SplitN` instances should actually be removed.

Comment (craciunoiuc) [reply internal/cli/kraft/cloud/instance/create/create.go:121]: If you also want to do safety checks, then yes, Split is fine

**Review by razvand — APPROVED** [review]

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**